core/src/main/scala/org/apache/spark/sql/store/StoreUtils.scala

```diff
@@ -428,6 +428,9 @@ object StoreUtils {
           if (hasOverflow) {
             overflowAdded = true
             s"$GEM_EVICTION_BY $v $GEM_OVERFLOW "
+          } else if (!parameters.contains(OVERFLOW)) {
+            // SNAP-1501 Set overflow as default evict-action unless specified otherwise.
+            s"$GEM_EVICTION_BY $v $GEM_OVERFLOW "
           } else {
             s"$GEM_EVICTION_BY $v "
           }
```

Comment:
Not needed with above mentioned change. Instead you can check that when EVICTION_BY is specified then hasOverflow has to be true i.e. if (hasOverflow) ... else throw Utils.analysisException(...)